product/src/main/java/com/yas/product/service/ProductRecommendationService.java

```diff
@@ -0,0 +1,156 @@
+@Transactional
+@Slf4j
+public class ProductRecommendationService {
+    private final ProductRepository productRepository;
+    private final MediaService mediaService;
+    private final ProductOptionCombinationRepository productOptionCombinationRepository;
+
+    /**
+     * Constructor for {@code ProductRecommendationService} that initializes the service with necessary dependencies.
+     *
+     * @param productRepository                  the repository responsible for handling product data
+     * @param mediaService                       the service responsible for managing media assets associated
+     *                                           with products
+     * @param productOptionCombinationRepository the repository for managing product option combinations
+     */
+    public ProductRecommendationService(ProductRepository productRepository, MediaService mediaService,
+                                        ProductOptionCombinationRepository productOptionCombinationRepository) {
+        this.productRepository = productRepository;
+        this.mediaService = mediaService;
+        this.productOptionCombinationRepository = productOptionCombinationRepository;
+    }
+
+    /**
+     * Retrieves detailed information about a product by its unique identifier.
+     *
+     * @param productId the unique identifier of the product to be retrieved
+     * @return a {@link ProductDetailInfoVm} containing detailed information about the product
+     * @throws NotFoundException if no product is found with the given {@code productId}
+     */
+    public ProductDetailInfoVm getProductDetailById(long productId) {
+        List<ProductVariationGetVm> variations = new ArrayList<>();
+        Product product = productRepository
+                .findById(productId)
+                .filter(Product::isPublished)
+                .orElseThrow(() ->
+                        new NotFoundException(Constants.ErrorCode.PRODUCT_NOT_FOUND, productId)
+                );
+
+        List<Category> categories = Optional.ofNullable(product.getProductCategories())
+                .orElse(Collections.emptyList())  // Handle null case
+                .stream()
+                .map(ProductCategory::getCategory)
+                .toList();
+
+        Long brandId = Optional.ofNullable(product.getBrand())
+                .map(Brand::getId)
+                .orElse(null);
+        String brandName = Optional.ofNullable(product.getBrand())
+                .map(Brand::getName)
+                .orElse(null);
+
+        List<ProductAttributeValueGetVm> productAttributes = product.getAttributeValues()
+                .stream()
+                .map(ProductAttributeValueGetVm::fromModel)
+                .toList();
+
+        if (Boolean.TRUE.equals(product.isHasOptions())) {
+            List<Product> productVariations = product.getProducts()
+                    .stream()
+                    .toList();
+            variations = productVariations.stream()
```

Comment:
Product variations are considered separate products, but the recommendation service expects to receive a product corresponding to a `long productId`. Is there any specific reason to retrieve variations along with the requested product? Let's discuss.

Reply:
Temporarily hold this discussion. Product variations will be handled in [#1149](https://github.com/nashtech-garage/yas/issues/1149), then this API/service will no longer be needed.